File: frontend/src/api/businessCluster.ts
import type { AxiosRequestConfig } from 'axios';
import api from './index';
import type { ApiResponse, PaginatedResponse, PaginationParams } from '../types/api';

export interface BusinessCluster {
  id: string;
  tenant_id: string;
  instance_id: string;
  name: string;
  display_name: string;
  kubeconfig_path: string;
  agent_status: string;
  log_agent_status: string;
  log_instance_id?: string;
  labels: string;
  metrics_collect_config?: string;
  logs_collect_config?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateBusinessClusterRequest {
  instance_id: string;
  name: string;
  display_name?: string;
  kubeconfig?: string;
  kubeconfig_path?: string;
  labels?: Record<string, string>;
  metrics_collect_config?: MetricsCollectConfig;
  logs_collect_config?: LogsCollectConfig;
}

export interface MetricsCollectConfig {
  select_all_by_default?: boolean;
  scrape_interval?: string;
  scrape_timeout?: string;
  namespace_include?: string[];
  namespace_exclude?: string[];
}

export interface LogsCollectConfig {
  namespace_include?: string[];
  namespace_exclude?: string[];
  exclude_paths?: string[];
}

export interface CollectConfigView {
  metrics: MetricsCollectConfig;
  logs: LogsCollectConfig;
}

export const businessClusterAPI = {
  list: (
    params?: PaginationParams & { workspace_id?: string; instance_id?: string },
    config?: AxiosRequestConfig,
  ) =>
    api.get<ApiResponse<PaginatedResponse<BusinessCluster>>>('/business-clusters', { ...config, params }),

  get: (id: string, config?: AxiosRequestConfig) =>
    api.get<ApiResponse<BusinessCluster>>(`/business-clusters/${id}`, config),

  create: (data: CreateBusinessClusterRequest) =>
    api.post<ApiResponse<BusinessCluster>>('/business-clusters', data),

  delete: (id: string, force?: boolean) =>
    api.delete<ApiResponse<null>>(`/business-clusters/${id}`, { params: { force: force ? 'true' : undefined } }),

  enableLogs: (id: string, data: { log_instance_id: string }) =>
    api.post<ApiResponse<BusinessCluster>>(`/business-clusters/${id}/enable-logs`, data),

  disableLogs: (id: string, force?: boolean) =>
    api.post<ApiResponse<BusinessCluster>>(`/business-clusters/${id}/disable-logs`, undefined, {
      params: { force: force ? 'true' : undefined },
    }),

  getCollectConfig: (id: string) =>
    api.get<ApiResponse<CollectConfigView>>(`/business-clusters/${id}/collect-config`),

  updateCollectConfig: (id: string, data: { metrics?: MetricsCollectConfig; logs?: LogsCollectConfig }) =>
    api.put<ApiResponse<CollectConfigView>>(`/business-clusters/${id}/collect-config`, data),
};

File: backend/internal/model/business_cluster.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// BusinessCluster 租户接入的业务集群（被监控目标）。
type BusinessCluster struct {
	ID              uuid.UUID      `json:"id" gorm:"type:uuid;primaryKey"`
	TenantID        uuid.UUID      `json:"tenant_id" gorm:"type:uuid;not null;index"`
	InstanceID      uuid.UUID      `json:"instance_id" gorm:"type:uuid;not null;index"`
	Name            string         `json:"name" gorm:"type:varchar(255);not null"`
	DisplayName     string         `json:"display_name" gorm:"type:varchar(255)"`
	Kubeconfig      string         `json:"-" gorm:"type:text"`
	KubeconfigPath  string         `json:"kubeconfig_path" gorm:"type:varchar(500)"`
	AgentStatus     string         `json:"agent_status" gorm:"type:varchar(20);default:pending"`
	LogAgentStatus  string         `json:"log_agent_status" gorm:"type:varchar(20);default:pending"`
	LogInstanceID   *uuid.UUID     `json:"log_instance_id" gorm:"type:uuid;index"`
	Labels          string         `json:"labels" gorm:"type:jsonb;default:'{}'"`
	// MetricsCollectConfig VMAgent 采集配置 JSON（见 MetricsCollectConfig）。
	MetricsCollectConfig string `json:"metrics_collect_config" gorm:"type:jsonb;default:'{}'"`
	// LogsCollectConfig Vector 采集配置 JSON（见 LogsCollectConfig）。
	LogsCollectConfig string `json:"logs_collect_config" gorm:"type:jsonb;default:'{}'"`
	CreatedAt         time.Time      `json:"created_at"`
	UpdatedAt         time.Time      `json:"updated_at"`
	DeletedAt         gorm.DeletedAt `json:"-" gorm:"index"`
}

func (BusinessCluster) TableName() string { return "ops_business_clusters" }

func (b *BusinessCluster) BeforeCreate(tx *gorm.DB) error {
	if b.ID == uuid.Nil {
		b.ID = uuid.New()
	}
	return nil
}
